src/puppetlabs/puppetdb/command.clj

```diff
@@ -235,16 +235,16 @@
 (defn store-report*
   [version db {:keys [payload annotations]}]
   (let [id          (:id annotations)
-        report      (upon-error-throw-fatality
-                     (report/validate! version payload))
-        certname    (:certname report)
+        {:keys [certname puppet_version] :as report} (->> payload
+                                                          (s/validate report/report-schema)
```

Comment:
I could think of a view choices here, for instance we're already validating at `add-report`. My main reason for keeping it was I'm not quite sure what `upon-error-throw-fatality` does, and my other reason is the we basically do the same thing for catalogs via `parse-catalog`.